src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubListener.cs

```diff
@@ -71,6 +72,8 @@ void IListener.Cancel()
 
         void IDisposable.Dispose()
         {
+            _disposed = true;

```

Comment:
I think you'll need to call `Dispose()` on the `EventProcessor` here as well. It already implements `IDisposable`, but I don't think anyone ever calls it (it's not registered with DI). That would let you track `IsDisposed` internally there, without the new interface.

Reply:
we do not create `EventProcessor` explicitly - `EventHubListener` can create multiple `EventProcessors` [here](https://github.com/Azure/azure-functions-eventhubs-extension/blob/5eb8543e47b448eb3b485a571a2ceb1ab404650f/src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubListener.cs) for each partition. Furthermore `EventProcessor `can be opened/closed for a partition multiple times during `EventHubListener` lifecycle due partition ownership rebalancing. So the approach you proposed will require to manage collection of active `EventProcessors `in `EventHubListener `but passing `EventHubListener `to each EventProcessors looks much more simpler.

Reply:
Yeah, I see now. That lifetime isn't clear and it doesn't look like EH calls Dispose() on any EventProcessors that it createss. I think what you've got is okay in that case.